odoo/fields.py

```diff
@@ -2191,8 +2209,41 @@ def _setup_attrs(self, model, name):
                     "%s: selection_add=%r must be a list" % (self, selection_add)
                 assert values is not None, \
                     "%s: selection_add=%r on non-list selection %r" % (self, selection_add, self.selection)
+
+                ondelete = field.args.get('ondelete') or {}
+                new_values = [kv[0] for kv in selection_add if kv[0] not in values]
+                for key in new_values:
+                    ondelete.setdefault(key, 'set null')
```

Comment:
Doesn't that mean there's always a value and the code in ir_model has no need to check for a missing / empty ondelete?

Reply:
Computed selection fields will have their ondelete set to `None` (see above), if they don't define a `selection_add` this will remain as `None` and they will be treated the same as non-compute selection fields during `_process_ondelete`